SourcingTests/CollectionViewDataSourceTest.swift

```diff
@@ -182,5 +182,21 @@ class CollectionViewDataSourceTest: XCTestCase {
         XCTAssertEqual(secondCollectionViewMock.reloadedCount, 1)
     }
 
-
+    func testMoveIndexPaths() {
+        //Given
+        let cellConfig = CellConfiguration<MockCollectionCell<Int>>(cellIdentifier: cellIdentifier)
+        let realCollectionView = UICollectionView(frame: .zero, collectionViewLayout: UICollectionViewLayout())
+        
+        //When
+        let dataSource = CollectionViewDataSource(collectionView: collectionViewMock, dataProvider: dataProvider, cellDequeable: cellConfig)
+        let fromIndexPath = IndexPath(row: 0, section: 1)
+        let toIndexPath = IndexPath(row: 1, section: 0)
+        dataProvider.moveItemAt(sourceIndexPath: fromIndexPath, to: toIndexPath)
```

Comment:
We should not call ```dataProvider.moveItemAt``` here. Instead we should call ```dataSource.moveItem```

The reason is that we want to test the dataSource and not the dataProvider